IconTemplates.lua

```diff
@@ -0,0 +1,183 @@
+local _, NS = ...;
+
+NS.ShowOverlayGlow = function (button)
+    if not button.spellActivationAlert then
+        return;
+    end
+
+    if button.spellActivationAlert.animOut:IsPlaying() then
+        button.spellActivationAlert.animOut:Stop();
+    end
+
+    if not button.spellActivationAlert:IsShown() then
+        button.spellActivationAlert.animIn:Play();
+    end
+end
+
+NS.HideOverlayGlow = function (button)
+    if not button.spellActivationAlert then
+        return;
+    end
+
+    if button.spellActivationAlert.animIn:IsPlaying() then
+        button.spellActivationAlert.animIn:Stop();
+    end
+
+    if button:IsVisible() then
+        button.spellActivationAlert.animOut:Play();
+    else
+        button.spellActivationAlert.animOut:OnFinished();	--We aren't shown anyway, so we'll instantly hide it.
+    end
+end
+
+local function OnCooldownTimerFinished(self)
+    local icon = self:GetParent();
+    if icon.group then
+        NS.IconGroup_Remove(icon:GetParent(), icon);
+    end
+end
+
+local function OnDurationTimerFinished(self)
+    local icon = self:GetParent();
+    NS.HideOverlayGlow(icon);
+    if icon.cooldown then
+        icon.cooldown:Show();
+    else
+        if icon.group then
+            NS.IconGroup_Remove(icon:GetParent(), icon);
+        end
+    end
+end
+
+NS.CreateWeakAuraIcon = function (unit, spellID, size, group)
+    local frame = CreateFrame("Frame", nil, UIParent);
+    frame:Hide();
+
+    frame.unit = unit;
+    frame.spellID = spellID;
+    frame.spell = NS.spellData[spellID];
+    frame.priority = frame.spell.priority;
+    frame.group = group;
+    frame:SetSize(size, size);
+
+    frame.tex = frame:CreateTexture();
+    frame.tex:SetTexture(select(3, GetSpellInfo(spellID)));
+    frame.tex:SetAllPoints();
+
+    -- Create duration/cooldown timers as needed
+    local spell = frame.spell;
+    if spell.cooldown then
+        frame.cooldown = CreateFrame("Cooldown", nil, frame, "CooldownFrameTemplate");
+        frame.cooldown:SetAllPoints();
+        frame.cooldown:SetDrawEdge(false);
+        frame.cooldown:SetDrawBling(false);
+        frame.cooldown:SetDrawSwipe(true);
+        frame.cooldown:SetReverse(true);
+        frame.cooldown:SetScript("OnCooldownDone", OnCooldownTimerFinished);
+
+        if spell.charges then
+            spell.chargeExpire = GetTime(); -- Set charge expirationTime
```

Comment:
1x SweepyBoop/IconTemplates.lua:111: attempt to compare nil with number
[string "@SweepyBoop/IconTemplates.lua"]:111: in function `StartWeakAuraIcon'
[string "@SweepyBoop/ArenaEnemyCooldown.lua"]:148: in function <SweepyBoop/ArenaEnemyCooldown.lua:59>
[string "@SweepyBoop/ArenaEnemyCooldown.lua"]:187: in function <SweepyBoop/ArenaEnemyCooldown.lua:185>

This should be set to frame.cooldown

Reply:
LUA on monk images

Reply:
Fixed in latest iteration